src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -99,32 +100,39 @@ async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)>
             var indicesResult = await TryGetIndicesAsync(
                 document.Project, forceIndexCreation, cancellationToken).ConfigureAwait(false);
 
-            // Don't show unimported extension methods if the index isn't ready.
-            if (!indicesResult.HasResult)
+            // Queue a background task to construct all requested indices in case we are missing any.
+            // But we still want to return what we have found, even if it's incomplete.
+            if (!indicesResult.HasFullResults)
             {
-                // We use a very simple approach to build the cache in the background:
-                // queue a new task only if the previous task is completed, regardless of what
-                // that task is.
+                Debug.Assert(!forceIndexCreation);
+
+                // For simplicity, we queue a new task only if the previous task is completed, 
+                // regardless of what that task is, and we don't distinguish missing indices
+                // from existing ones.
                 lock (s_gate)
                 {
                     if (s_indexingTask.IsCompleted)
                     {
-                        s_indexingTask = Task.Run(() => TryGetIndicesAsync(document.Project, forceIndexCreation: true, CancellationToken.None));
+                        s_indexingTask = Task.Run(() => PopulateCacheEntries(document.Project));
                     }
                 }
```

Comment:
if the indexing task is complete... why do we need a new task?

Reply:
This extremely simple approach we are using here doesn't keep track of what the previous task was doing, it could be a different set of references it was calculating. We got here because the index of at least one reference is missing, and we don't want to try to access the index again (and likely fail again).

Reply:
i see.  that's fine :)  i just didn't really understand what was going on.